pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -201,5 +203,26 @@ func (b *MacvtapLibvirtSpecGenerator) generate() error {
 }
 
 func (b *MacvtapLibvirtSpecGenerator) discoverDomainIfaceSpec() (*api.Interface, error) {
-	return &b.cachedDomainInterface, nil
+	var domainIface api.Interface
+
+	podNicLink, err := b.handler.LinkByName(b.podInterfaceName)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get a link for interface: %s", b.podInterfaceName)
+		return nil, err
+	}
+	mac, err := virtnetlink.RetrieveMacAddressFromVMISpecIface(b.vmiSpecIface)
+	if err != nil {
+		return nil, err
+	}
+	if mac == nil {
+		mac = &podNicLink.Attrs().HardwareAddr
+	}
+
+	domainIface.MAC = &api.MAC{MAC: mac.String()}
+	domainIface.MTU = &api.MTU{Size: strconv.Itoa(podNicLink.Attrs().MTU)}
+	domainIface.Target = &api.InterfaceTarget{
+		Device:  b.podInterfaceName,
+		Managed: "no",
+	}
+	return &domainIface, nil
```

Comment:
this can be simplified:

```golang
return &api.Interface{
    MAC:    &api.MAC{MAC: mac.String()}
    MTU:    &api.MTU{Size: strconv.Itoa(podNicLink.Attrs().MTU)}
    Target: &api.InterfaceTarget{
		Device:  b.podInterfaceName,
		Managed: "no",
	}
}, nil
```